kata-webhook/main.go

```diff
@@ -44,22 +44,24 @@ func annotatePodMutator(_ context.Context, obj metav1.Object) (bool, error) {
 	for i := range pod.Spec.Containers {
 		if pod.Spec.Containers[i].SecurityContext != nil {
 			if *pod.Spec.Containers[i].SecurityContext.Privileged {
-				fmt.Println("Privileged container: ", pod.GetNamespace(), pod.GetName())
+				fmt.Println("privileged container: ", pod.GetNamespace(), pod.GetName())
 				return false, nil
 			}
 		}
 	}
 
-	fmt.Println("katait: ", pod.GetNamespace(), pod.GetName())
+	if pod.Spec.RuntimeClassName != nil {
+		fmt.Println("explicit runtime: ", pod.GetNamespace(), pod.GetName(), pod.Spec.RuntimeClassName)
+		return false, nil
+	}
+
+	// Mutate the pod
+	fmt.Println("setting runtime to kata: ", pod.GetNamespace(), pod.GetName())
 
-	// Mutate our object with the required annotations.
-	if pod.Annotations == nil {
-		pod.Annotations = make(map[string]string)
+	kataRuntimeClassName := "kata"
+	if pod.Spec.RuntimeClassName == nil {
```

Comment:
This must be true as you checked it above?

Reply:
fixed